### PR TITLE
Blind counting sort

### PR DESCRIPTION
**Pull request by filedesless**
New algorithm based on the counting sort, scales linearly with the number of elements to sort.

- [x] blind_array_add
- [x] fix correctness
- [x] improve performance (LUT::from_lwe bottleneck)
  - [x] blind_array_inc (blind_array_add_trivial?) special case could be faster (use LUT::from_vec_trivially instead of LUT::from_lwe)
  - [x] improve LUT::from_lwe (parallelism? need tfhe 0.6)
  - [ ] ~replace LUT::from_lwe creation and evaluation by a cmux?~
- [x] investigate flakiness

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1056,10 +1028,6 @@ impl PublicKey {
             result_glwe = self.glwe_sum(&result_glwe, &many_lut[i].0);
         }
```
Comment (sofianeazogagh) [inline src/lib.rs:1029]: Why the rotation of half_box_size is removed here ? 

        // Rotation of -half_box_size to get the accurate result when we sample extract
        let poly_monomial_degree = MonomialDegree(2 * ctx.polynomial_size().0 - ctx.box_size() / 2);
        self.glwe_absorption_monic_monomial(&mut result_glwe, poly_monomial_degree);

File: src/lib.rs [hunk 2]
```diff
@@ -1306,6 +1274,33 @@ impl PublicKey {
         programmable_bootstrap_lwe_ciphertext(&ct_input, &mut lwe, &lut.0, &self.fourier_bsk);
         self.allocate_and_keyswitch_lwe_ciphertext(lwe, ctx)
     }
+
+    /// blindly adds x to the i-th coefficient of the given LUT
+    pub fn blind_array_add(
```
Comment (sofianeazogagh) [inline src/lib.rs]: Suggestion : rename blind_array_add by blind_array_inject ?

File: src/lib.rs [hunk 3]
```diff
@@ -1371,15 +1366,6 @@ impl LUT {
         redundant_many_lwe
     }
```
Comment (sofianeazogagh) [inline src/lib.rs:1369]: Why this function is removed ?

    fn add_redundancy(
        lwe: &LweCiphertext<Vec<u64>>,
        ctx: &Context,
    ) -> Vec<LweCiphertext<Vec<u64>>> {
        let box_size = ctx.box_size();
        let redundant_lwe: Vec<LweCiphertext<Vec<u64>>> = vec![(*lwe).clone(); box_size];
        redundant_lwe
    }

File: src/lib.rs (hunk 1 above)
Comment (filedesless) [reply src/lib.rs:1029]: because `to_many_lut` has been changed to use `LUT::from_lwe` which centers the boxes already

File: src/lib.rs (hunk 3 above)
Comment (filedesless) [reply src/lib.rs:1369]: because it's dead code

**Review by sofianeazogagh — APPROVED** [review]

**Pull request merged**